systemd/coreos-installer-generator

```diff
@@ -17,7 +17,20 @@ karg() {
     echo "${value}"
 }
 
+karg_bool() {
+    local value=$(karg "$@")
+    case "$value" in
+        ""|0|no|off) return 1;;
+        *) return 0;;
+    esac
+}
+
 if [ -n "$(karg coreos.inst.install_dev)" ]; then
     ln -sf "/usr/lib/systemd/system/coreos-installer.target" \
         "${UNIT_DIR}/default.target"
+
+    # Create precondition for coreos-installer-reboot.service if requested
+    if ! karg_bool coreos.inst.skip_reboot; then
+        touch /run/coreos-installer-reboot
```

Comment:
Hmm, I'm confused why we need the reboot to happen in a separate service entirely. Can't we just do it from the installer directly after a successful install? E.g. we add a `--reboot` flag and have `coreos-installer-service` just pass that in if the karg is specified.

Reply:
I think @bgilbert was trying to keep the utility separate from the service, but I'm not opposed to adding a --reboot flag to the utility. Though maybe we can open a separate issue and do that in a followup so we don't block?